src/python/pants/engine/internals/build_files.py

```diff
@@ -221,7 +233,13 @@ async def find_target_adaptor(request: TargetAdaptorRequest) -> TargetAdaptor:
             "Generated targets are not defined in BUILD files, and so do not have "
             f"TargetAdaptors: {request}"
         )
-    address_family = await Get(AddressFamily, AddressFamilyDir(address.spec_path))
+    address_family = await Get(
+        AddressFamily,
+        AddressFamilyDir(
+            address.spec_path,
+            ignore_unrecognized_build_file_symbols=request.ignore_unrecognized_build_file_symbols,
+        ),
+    )
```

Comment:
One other thought here is that you could potentially use:

```suggestion
    address_family = await Get(
        AddressFamily,
        {
             AddressFamilyDir(address.spec_path): AddressFamilyDir,
            IgnoreUnrecognizedBuildFileSymbols(true): IgnoreUnrecognizedBuildFileSymbols,
        }
    )
```

That would effectively be equivalent to making it a global `Param` or scheduler singleton though.